StRoot/StFstClusterMaker/StFstScanRadiusClusterAlgo.cxx

```diff
@@ -0,0 +1,238 @@
+#include "StEvent.h"
+#include "StEvent/StEnumerations.h"
+#include "StMessMgr.h"
+#include "StFstScanRadiusClusterAlgo.h"
+#include "StFstUtil/StFstCollection.h"
+#include "StFstUtil/StFstRawHitCollection.h"
+#include "StFstUtil/StFstRawHit.h"
+#include "StFstUtil/StFstClusterCollection.h"
+#include "StFstUtil/StFstCluster.h"
+#include "StEvent/StFstConsts.h"
+
+#include <math.h>
+#include <iostream>
+#include <iterator>
+#include <vector>
+#include <algorithm>
+
+
+Int_t StFstScanRadiusClusterAlgo::doClustering(const StFstCollection &fstCollection, StFstRawHitCollection &rawHitsOriginal, StFstClusterCollection &clusters )
+{
+  StFstCluster *newCluster = 0;
+  StFstRawHit *rawHitTemp = 0;
+  StFstRawHit *rawHitMaxAdcTemp = 0;
+  Int_t clusterType = kFstScanRadiusClusterAlgo;
+  Int_t maxTb = -1, usedTb = -1;
+  Int_t disk = 0, wedge = 0, sensor = -1, apv = -1;
+  Float_t meanRStrip = -1., maxRStrip = -1, meanPhiStrip = -1;
+  Float_t totCharge = 0., totChargeErr = 0.;
+  Int_t clusterSize = 0, clusterSizeR = 0, clusterSizePhi = 0;
+  unsigned short idTruth = 0;
+
+  //get number of time bin used in this event
+  Int_t nTimeBins = fstCollection.getNumTimeBins();
+
+  //sort raw hits in increasing order by geometry ID
+  rawHitsOriginal.sortByGeoId();
+
+  //copy raw hit collection to temporary vectors
+  std::vector<StFstRawHit *>  rawHitsVec[kFstNumSensorsPerWedge][kFstNumPhiSegPerSensor];
+  std::vector<StFstCluster *> clustersVec[kFstNumSensorsPerWedge][kFstNumPhiSegPerSensor];
+  std::vector<StFstRawHit *>  rawHitsToMerge;
+
+  for (int sensorIdx = 0; sensorIdx < kFstNumSensorsPerWedge; sensorIdx++) {
+    for (int phiIdx = 0; phiIdx < kFstNumPhiSegPerSensor; phiIdx++) {
+      rawHitsVec[sensorIdx][phiIdx].reserve(kFstNumRStripsPerSensor);
+      clustersVec[sensorIdx][phiIdx].reserve(kFstNumRStripsPerSensor);
+    }
+  }
+
+  rawHitsToMerge.reserve(kFstNumRStripsPerSensor);
+
+  for (std::vector< StFstRawHit * >::iterator rawHitPtr = rawHitsOriginal.getRawHitVec().begin(); rawHitPtr != rawHitsOriginal.getRawHitVec().end(); ++rawHitPtr) {
+    int sensorIndex = (int)(*rawHitPtr)->getSensor();
+    int phiIndex = (int)(*rawHitPtr)->getPhiStrip();
+    rawHitsVec[sensorIndex][phiIndex].push_back( new StFstRawHit( *(*rawHitPtr)) );
+  }
+
+  //do clustering
+  int clusterLabel = 0;
+
+  for (int sensorIdx = 0; sensorIdx < kFstNumSensorsPerWedge; sensorIdx++)
+  {
+    //step 1: do clustering for each phi
+    for (int phiIdx = 0; phiIdx < kFstNumPhiSegPerSensor; phiIdx++)
+    {
+      while ( !rawHitsVec[sensorIdx][phiIdx].empty() )
+      {
+	rawHitTemp = rawHitsVec[sensorIdx][phiIdx].back();
+	rawHitsVec[sensorIdx][phiIdx].pop_back();
+	rawHitsToMerge.push_back(rawHitTemp);
+	//count number to merge
+	int nToMerge = 1;
+	//find all raw hits that are neighboring
+	std::vector<StFstRawHit *>::iterator rawHitsToMergePtr = rawHitsToMerge.begin();
+	rawHitMaxAdcTemp = *rawHitsToMergePtr;
+	// StFstRawHit *rawHitNext = 0;
```

Comment:
Please remove commented not used statements

Reply:
done in a1b656422171901b3c0f9af39228875d9fe1be5e